samples/snippets/sample_custom_types.py

```diff
@@ -0,0 +1,525 @@
+
+    Return:
+        Instance object.
+    """
+    assert cpu_series in (
+        CustomMachineType.CPUSeries.E2_MICRO,
+        CustomMachineType.CPUSeries.E2_SMALL,
+        CustomMachineType.CPUSeries.E2_MEDIUM,
+    )
+    custom_type = CustomMachineType(zone, cpu_series, memory)
+    return create_instance(project_id, zone, instance_name, custom_type)
+
+
+# [END compute_custom_machine_type_create_shared_with_helper ]
+
+
+# [START compute_custom_machine_type_create_without_helper ]
+def create_custom_machines_no_helper(
+    project_id: str, zone: str, instance_name: str, core_count: int, memory: int
+):
+    """
+    Create new VM instances without using a CustomMachineType helper function.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone: name of the zone to create the instance in. For example: "us-west3-b"
+        instance_name: name of the new virtual machine (VM) instance.
+        core_count: Number of CPU cores you want to use.
+        memory: Number of megabytes of memory you want to use.
+
+    Returns:
+        List of Instance objects.
+    """
+    # The core_count and memory values are not validated anywhere and can be rejected by the API.
+    instances = [
+        create_instance(
+            project_id,
+            zone,
+            f"{instance_name}_n1",
+            f"zones/{zone}/machineTypes/custom-{core_count}-{memory}",
+        ),
+        create_instance(
+            project_id,
+            zone,
+            f"{instance_name}_n2",
+            f"zones/{zone}/machineTypes/n2-custom-{core_count}-{memory}",
+        ),
+        create_instance(
+            project_id,
+            zone,
+            f"{instance_name}_n2d",
+            f"zones/{zone}/machineTypes/n2d-custom-{core_count}-{memory}",
+        ),
+        create_instance(
+            project_id,
+            zone,
+            f"{instance_name}_e2",
+            f"zones/{zone}/machineTypes/e2-custom-{core_count}-{memory}",
+        ),
+        create_instance(
+            project_id,
+            zone,
+            f"{instance_name}_e2_micro",
+            f"zones/{zone}/machineTypes/e2-custom-micro-{memory}",
+        ),
+        create_instance(
+            project_id,
+            zone,
+            f"{instance_name}_e2_small",
+            f"zones/{zone}/machineTypes/e2-custom-small-{memory}",
+        ),
+        create_instance(
+            project_id,
+            zone,
+            f"{instance_name}_e2_medium",
+            f"zones/{zone}/machineTypes/e2-custom-medium-{memory}",
+        ),
+    ]
+    return instances
+
+
+# [END compute_custom_machine_type_create_without_helper ]
+
+
+# [START compute_custom_machine_type_extra_mem_no_helper ]
+def create_custom_machines_extra_mem_no_helper(
+    project_id: str, zone: str, instance_name: str, core_count: int, memory: int
+):
+    """
+    Create new VM instances without using a CustomMachineType helper function.
+
```

Comment:
How is this function different from the previous one? Is it using extended memory for the VM? If yes, can we rewrite this comment accordingly?

Reply:
Yes, the extended memory is the difference. But also, it is not using the helper class, but creates the custom machine type string manually.